src/orion/algo/nevergradoptimizer.py

```diff
@@ -0,0 +1,362 @@
+        raise NotImplementedError(
+            "All categories in Categorical must have the same probability."
+        )
+    return ng.p.Choice(dim.interval())
+
+
+@to_ng_space.register("real", "uniform")
+def _(_, dim):
+    lower, upper = dim.interval()
+    if dim.shape:
+        # Temporary fix pending [#800]
+        # ng.p.Array expects an array with a shape or a float
+        # an array with an empty shape is not valid
+        # so we cast it to a float
+        if hasattr(lower, "shape") and lower.shape == ():
+            lower = float(lower)
+
+        if hasattr(upper, "shape") and upper.shape == ():
+            upper = float(upper)
+
+        return ng.p.Array(lower=lower, upper=upper, shape=_intshape(dim.shape))
+    else:
+        return ng.p.Scalar(lower=lower, upper=upper)
+
+
+@to_ng_space.register("integer", "int_uniform")
+def _(self, dim):
+    return self["real", "uniform"](self, dim).set_integer_casting()
+
+
+@to_ng_space.register("real", "reciprocal")
+def _(_, dim):
+    if dim.shape:
+        raise NotImplementedError("Array with reciprocal prior cannot be converted.")
+    lower, upper = dim.interval()
+    return ng.p.Log(lower=lower, upper=upper, exponent=2)
+
+
+@to_ng_space.register("integer", "int_reciprocal")
+def _(self, dim):
+    return self["real", "reciprocal"](self, dim).set_integer_casting()
+
+
+@to_ng_space.register("real", "norm")
+def _(_, dim):
+    if dim.shape:
+        raise NotImplementedError("Array with normal prior cannot be converted.")
+    return ng.p.Scalar(init=dim.original_dimension.prior.mean()).set_mutation(
+        sigma=dim.original_dimension.prior.std()
+    )
+
+
+@to_ng_space.register("real", "normal")
+def _(_, dim):
+    if dim.shape:
+        raise NotImplementedError("Array with normal prior cannot be converted.")
+    return ng.p.Scalar(init=dim.original_dimension.prior.mean()).set_mutation(
+        sigma=dim.original_dimension.prior.std()
+    )
+
+
+@to_ng_space.register("fidelity", "None")
+def _(_, dim):
+    if dim.shape:
+        raise NotImplementedError("Array of Fidelity cannot be converted.")
+    _, upper = dim.interval()
+    # No equivalent to Fidelity space, so we always use the upper value
+    return upper
+
+
+NOT_WORKING = {
+    "BO",
+    "BOSplit",
+    "BayesOptimBO",
+    "DiscreteDoerrOnePlusOne",
+    "NelderMead",
+    "PCABO",
+    "SPSA",
+    "SQP",
+    "NoisyBandit",
+    "NoisyOnePlusOne",
+    "OptimisticDiscreteOnePlusOne",
+    "OptimisticNoisyOnePlusOne",
+    "Powell",
+    "CmaFmin2",
+    "RPowell",
+    "RSQP",
+    "PymooNSGA2",
+}
+
+
+class NevergradOptimizer(BaseAlgorithm):
+    """Wraps the nevergrad library to expose its algorithm to orion
+
+    Parameters
+    ----------
+    space: `orion.algo.space.Space`
+        Optimisation space with priors for each dimension.
+    seed: None, int or sequence of int
+        Seed for the random number generator used to sample new trials.
+        Default: ``None``
```

Comment:
Missing model_name, budget and num_workers.

Reply:
added